### PR TITLE
Truncate Summary and Send Version to Bugzilla

### PR DESCRIPTION
**Pull request by coreydaley**
- Truncate the summary to 255 characters when opening a Bugzilla
 - Send the version when opening a Bugzilla

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -725,7 +725,7 @@ func (s *Server) printHtmlReport(w http.ResponseWriter, req *http.Request) {
 		html.WriteLandingPage(w, s.options.Releases)
 		return
 	}
-	html.PrintHtmlReport(w, req, s.analyzers[release].Report, s.analyzers[release+"-prev"].Report, s.options.EndDay, 15)
+	html.PrintHtmlReport(w, req, s.analyzers[release].Report, s.analyzers[release+"-prev"].Report, s.options.EndDay, 15, release)
```
Comment (bparees) [inline main.go:728]: analyzers[release].Report already has a Release field in it, you shouldn't need to pass this arg in.

File: pkg/html/html.go [hunk 2]
```diff
@@ -385,7 +385,7 @@ func summaryTopFailingTests(topFailingTestsWithoutBug, topFailingTestsWithBug []
...
-			bug = fmt.Sprintf("<a target=\"_blank\" href=https://bugzilla.redhat.com/enter_bug.cgi?classification=Red%%20Hat&product=OpenShift%%20Container%%20Platform&cf_internal_whiteboard=buildcop&short_desc=%[1]s&cf_environment=%[1]s&comment=test:%%0A%[1]s%%20%%0A%%0Ais%%20failing%%20frequently%%20in%%20CI,%%20see%%20search%%20results:%%0A%s%%0A%%0A%s>Open a bug</a>", url.QueryEscape(test.Name), url.QueryEscape(searchUrl), url.QueryEscape(exampleJob))
+			bug = fmt.Sprintf("<a target=\"_blank\" href=https://bugzilla.redhat.com/enter_bug.cgi?classification=Red%%20Hat&product=OpenShift%%20Container%%20Platform&cf_internal_whiteboard=buildcop&short_desc=%[1]s&cf_environment=%[1]s&comment=test:%%0A%[1]s%%20%%0A%%0Ais%%20failing%%20frequently%%20in%%20CI,%%20see%%20search%%20results:%%0A%s%%0A%%0A%s&version=%[1]s>Open a bug</a>", url.QueryEscape(test.Name[:255]), url.QueryEscape(searchUrl), url.QueryEscape(exampleJob), release)
```
Comment (bparees) [inline pkg/html/html.go:388]: this is truncating the test name everywhere in the bz template, you should only be truncating it in the summary field.

File: main.go (hunk 1 above)
Comment (coreydaley) [reply main.go]: fixed in current push

File: pkg/html/html.go (hunk 2 above)
Comment (coreydaley) [reply pkg/html/html.go]: fixed in current push

**Comment by bparees** [comment]
@coreydaley were you able to test this locally?

**Comment by coreydaley** [comment]
Yes, and I fixed an issue with the short description truncating just now.

File: pkg/html/html.go [hunk 3]
```diff
@@ -385,7 +385,11 @@ func summaryTopFailingTests(topFailingTestsWithoutBug, topFailingTestsWithBug []
...
 			exampleJob := "FIXME: Replace this paragraph with a particular job URI from the search results to ground discussion.  A given test may fail for several reasons, and this bug should be scoped to one of those reasons.  Ideally you'd pick a job showing the most-common reason, but since that's hard to determine, you may also chose to pick a job at random.  Release-gating jobs (release-openshift-...) should be preferred over presubmits (pull-ci-...) because they are closer to the released product and less likely to have in-flight code changes that complicate analysis."
-			bug = fmt.Sprintf("<a target=\"_blank\" href=https://bugzilla.redhat.com/enter_bug.cgi?classification=Red%%20Hat&product=OpenShift%%20Container%%20Platform&cf_internal_whiteboard=buildcop&short_desc=%[1]s&cf_environment=%[1]s&comment=test:%%0A%[1]s%%20%%0A%%0Ais%%20failing%%20frequently%%20in%%20CI,%%20see%%20search%%20results:%%0A%s%%0A%%0A%s>Open a bug</a>", url.QueryEscape(test.Name), url.QueryEscape(searchUrl), url.QueryEscape(exampleJob))
+			short_desc := url.QueryEscape(test.Name)
+			if len(short_desc) > 255 {
+				short_desc = short_desc[:255]
+			}
```
Comment (bparees) [inline pkg/html/html.go:391]: doesn't this risk truncating an escaped character and screwing everything up?

Comment (coreydaley) [reply pkg/html/html.go:391]: fixed

Comment (bparees) [reply pkg/html/html.go:391]: so the escape characters don't count towards the limit?  (makes sense i guess that BZ only sees the unescaped value)

**Pull request merged**